kloppy/infra/serializers/event/wyscout/deserializer_v3.py

```diff
@@ -110,24 +111,74 @@ def _flip_point(point: Point) -> Point:
     return Point(x=100 - point.x, y=100 - point.y)
 
 
-class ShotZoneResults(str, Enum):
-    GOAL_BOTTOM_LEFT = "glb"
-    GOAL_BOTTOM_RIGHT = "grb"
-    GOAL_BOTTOM_CENTER = "gb"
-    GOAL_CENTER_LEFT = "gl"
-    GOAL_CENTER = "gc"
-    GOAL_CENTER_RIGHT = "gr"
-    GOAL_TOP_LEFT = "glt"
-    GOAL_TOP_RIGHT = "grt"
-    GOAL_TOP_CENTER = "gt"
-    OUT_BOTTOM_RIGHT = "obr"
-    OUT_BOTTOM_LEFT = "olb"
-    OUT_RIGHT = "or"
-    OUT_LEFT = "ol"
-    OUT_LEFT_TOP = "olt"
-    OUT_TOP = "ot"
-    OUT_RIGHT_TOP = "ort"
-    BLOCKED = "bc"
+class ShotZoneResults(Enum):
+    """
+    Wyscout does not provide end-coordinates of shots. Instead shots on goal
+    are tagged with a zone. The zones and corresponding y and z-coordinates are depicted below.
+
+                                       (z-coordinate of zone or post)
+
+        olt      | ot |      ort          3.5
+     --------------------------------
+          ||=================||           2.77
+     -------------------------------
+          || glt | gt | grt ||            2
+     --------------------------------
+      ol || gl | gc  | gr || or           1
+     --------------------------------
+      olb || glb  | gb | grb || orb       0
+
+      40     45    50    55     60    (y-coordinate of zone)
+        44.62               55.38     (y-coordinate of post)
+    """
+
+    GOAL_BOTTOM_LEFT = ("glb", Point3D(100, 45, 0))
+    GOAL_BOTTOM_CENTER = ("gb", Point3D(100, 50, 0))
+    GOAL_BOTTOM_RIGHT = ("grb", Point3D(100, 55, 0))
+    GOAL_CENTER_LEFT = ("gl", Point3D(100, 45, 1))
+    GOAL_CENTER = ("gc", Point3D(100, 50, 1))
+    GOAL_CENTER_RIGHT = ("gr", Point3D(100, 55, 1))
+    GOAL_TOP_LEFT = ("glt", Point3D(100, 45, 2))
+    GOAL_TOP_CENTER = ("gt", Point3D(100, 50, 2))
+    GOAL_TOP_RIGHT = ("grt", Point3D(100, 55, 2))
+    POST_BOTTOM_LEFT = ("plb", Point3D(100, 44.62, 0))
+    POST_BOTTOM_RIGHT = ("pbr", Point3D(100, 55.38, 0))
+    OUT_BOTTOM_LEFT = ("olb", Point3D(100, 40, 0))
+    OUT_BOTTOM_RIGHT = ("obr", Point3D(100, 60, 0))
+    POST_LEFT = ("pl", Point3D(100, 44.62, 1))
+    POST_RIGHT = ("pr", Point3D(100, 55.38, 1))
+    OUT_LEFT = ("ol", Point3D(100, 40, 1))
+    OUT_RIGHT = ("or", Point3D(100, 60, 1))
+    POST_TOP_LEFT = ("ptl", Point3D(100, 44.62, 2))
+    POST_TOP_RIGHT = ("ptr", Point3D(100, 55.38, 2))
+    OUT_TOP_LEFT = ("otl", Point3D(100, 40, 3.5))
+    OUT_TOP_RIGHT = ("ort", Point3D(100, 60, 3.5))
+    POST_TOP = ("pt", Point3D(100, 50, 2.77))
+    OUT_TOP = ("ot", Point3D(100, 50, 3.5))
+    BLOCKED = ("bc", None)
+
+    def __init__(self, code: str, point: Point3D):
+        self._code = code
+        self._point = point
+
+    @property
+    def code(self) -> str:
+        """Returns the string identifier of the shot result."""
+        return self._code
+
+    @property
+    def point(self) -> Point3D:
+        """Returns the predefined Point3D associated with the shot result."""
+        return self._point
```

Comment:
Minor, but I think using properties here doesn't have any added value.
 ```suggestion
        self.code = code
        self.point = point
```